File: backend/Domains/Lesson.cs
namespace LumiLearn.Domains
{
    public class Lesson
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public Guid CourseId { get; set; }
        public DateTime Timestamp { get; set; } = DateTime.Now;

        // Navigation Properties
        public Course Course { get; set; }

        public ICollection<FlashCardSet> FlashCardSets { get; set; }
        public ICollection<Quiz> Quizzes { get; set; }
    }
}

File: frontend/types/lesson.ts
import { FlashcardSetOverview } from "./flashcardSet";
import { QuizzOverview } from "./quizz";

export type LessonOverview = {
    id: string;
    courseId: string;
    title: string;
    flashcardSets: FlashcardSetOverview[];
    quizzes: QuizzOverview[];
};

export type CreateLessonRequest = {
    id: string;
    title: string;
    courseId: string
}